tests/test_solidity_and_vyper.py

```diff
@@ -107,3 +108,40 @@ def test_dynamic_vyper_struct_arrays(vyper_contract):
     array_output = method_abi.outputs[0]
     assert array_output.type == "((address,bytes32),uint256)[]"
     assert array_output.canonical_type == "((address,bytes32),uint256)[]"
+
+
+def test_select_by_name(vyper_contract):
+    contract_type = ContractType.parse_obj(vyper_contract)
+    assert (
```

Comment:
Consider adding a more literal expected value here!

If every `__getitem__` call returned `None`, this would pass the test.